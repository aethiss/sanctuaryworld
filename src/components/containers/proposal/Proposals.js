import React from 'react'
import PropTypes from 'prop-types'

// Components
import BodyContainer from '../../commons/wrapper/BodyContainerResponsive'

// Styles
import { responsiveMap } from '../../commons/commonStyles/ResponsiveMap'
import ButtonDescription from '../../commons/buttons/buttonDescription/ButtonDescription'

const calculateTotalProposals = (proposalObj) => {
  let total = 0
  Object.keys(proposalObj).map((type) => {
    total += proposalObj[type].length
  })

  return total
}

const Proposals = ({ proposals }) => {
  const classes = responsiveMap('column', 'center', '/images/background/diablo4-proposal-bg.jpg')()
  return (
    <BodyContainer >
      <span className={classes.wrapperTop} />
      <div className={classes.wrapper}>
        <h3>Total Proposals : {calculateTotalProposals(proposals)}</h3>
        <ButtonDescription
          text='World & Story'
          anchorLink='/proposal/world'
        />
        <ButtonDescription
          text='Classes & PVE'
          anchorLink='/proposal/pve'
        />
        <ButtonDescription
          text='Trading & Crafting'
          anchorLink='/proposal/trading'
        />
        <ButtonDescription
          text='PVP & Combat'
          anchorLink='/proposal/pvp'
        />
      </div>
      <span className={classes.wrapperBottom} />
    </BodyContainer>
  )
}

Proposals.propTypes = {
  proposals: PropTypes.object,
}

export default Proposals
