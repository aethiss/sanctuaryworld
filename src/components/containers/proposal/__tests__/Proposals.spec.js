import React from 'react'
import TestRenderer from 'react-test-renderer'
import Proposals from '../Proposals'


describe('Proposals', () => {
  describe('Render', () => {
    it('[Proposals] : Render default', () => {
      const testRenderer = TestRenderer.create(<Proposals proposals={['ciao', 'pippo']} />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
