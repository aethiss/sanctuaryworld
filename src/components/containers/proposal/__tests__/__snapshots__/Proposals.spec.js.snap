// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposals Render [Proposals] : Render default 1`] = `
<div
  className="makeStyles-bodyWrapper-4"
>
  <span
    className="makeStyles-wrapperTop-2"
  />
  <div
    className="makeStyles-wrapper-1"
  >
    <h3>
      Total Proposals : 
      2
    </h3>
    <a
      href="/proposal/world"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-button-5"
      >
        <span
          className="makeStyles-buttonText-6"
        >
          World & Story
        </span>
      </button>
    </a>
    <a
      href="/proposal/pve"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-button-5"
      >
        <span
          className="makeStyles-buttonText-6"
        >
          Classes & PVE
        </span>
      </button>
    </a>
    <a
      href="/proposal/trading"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-button-5"
      >
        <span
          className="makeStyles-buttonText-6"
        >
          Trading & Crafting
        </span>
      </button>
    </a>
    <a
      href="/proposal/pvp"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-button-5"
      >
        <span
          className="makeStyles-buttonText-6"
        >
          PVP & Combat
        </span>
      </button>
    </a>
  </div>
  <span
    className="makeStyles-wrapperBottom-3"
  />
</div>
`;
